### PR TITLE
Remove id_tax_rules_group in carrier table

### LINKED ISSUE #25208: id_tax_rules_group in carrier table is not used anymore
#### Is your feature request related to a problem?

id_tax_rules_group in carrier table is not used anymore

#### Describe the solution you'd like

Remove the id_tax_rules_group  column from the carrier table

#### Additional context

* PrestaShop version: develop
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | id_tax_rules_group is not used anymore.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     |  Fixes https://github.com/PrestaShop/PrestaShop/issues/25208
| How to test?      | remove column in database without this pr and test an order and stock changes. 
| Possible impacts? | none

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25207)

### CODE REVIEW

**Comment by jf-viguier** [comment]
Is this column can be safely deleted or it should be a bc break ?

**Comment by Progi1984** [comment]
@jf-viguier Sorry... But I have another parts in code linked to carriers :
* `admin-dev/themes/default/template/controllers/carriers/helpers/form/form.tpl` (l.37)
* `controllers/admin/AdminCarriersController.php` a lot of `id_tax_rules_group`
* `js/admin/carrier_wizard.js` l.197
* `UI/pages/BO/shipping/carriers/add.js` l.26

**Comment by jf-viguier** [comment]
@Progi1984 id_tax_rules_group is stored for the carrier in another table than carrier table : 
 carrier_tax_rules_group_shop.id_tax_rules_group

**Comment by jf-viguier** [comment]
@Progi1984 thanks to automatic tests, I've removed id_tax_rules_group from initial data

**Comment by jf-viguier** [comment]
@Progi1984 all tests are ok now, please remove Waiting for author label as all id_tax_rules_group you founded are related to the other table, not carrier table.

**Review by Progi1984 — DISMISSED** [review]

**Comment by jf-viguier** [comment]
what's the problem with Integration tests (7.4) ?

**Comment by Progi1984** [comment]
@jf-viguier Restarted CI.

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Upgrade scripts need to be updated :)

**Comment by jf-viguier** [comment]
@PierreRambaud nice catch, added in 1.7.9 upgrade

File: install-dev/upgrade/sql/1.7.9.0.sql [hunk 1]
```diff
@@ -9,3 +9,4 @@ INSERT INTO `PREFIX_configuration` (`name`, `value`, `date_add`, `date_upd`) VAL
 ;
 INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
   (NULL, 'actionValidateOrderAfter', 'New Order', 'This hook is called after validating an order by core', '1');
+ALTER TABLE `PREFIX_carrier` DROP COLUMN `id_tax_rules_group`;
```
Comment (PierreRambaud) [inline install-dev/upgrade/sql/1.7.9.0.sql]:
```suggestion
ALTER TABLE `PREFIX_carrier` DROP COLUMN `id_tax_rules_group`;

```

**Review by PierreRambaud — DISMISSED** [review]

**Comment by jf-viguier** [comment]
Thanks @PierreRambaud

**Review by matthieu-rolland — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
@jf-viguier there is still a reference to this DB field here
https://github.com/PrestaShop/PrestaShop/blob/7c005444cf00710a81149e6c5ca861f32c600ce6/classes/Carrier.php#L606

Even if the function doesn't seem to be used by the core, but it might be used by modules So either we remove it (which is a BC break) or we can update the query to rely on the other table `carrier_tax_rules_group_shop`

**Comment by jf-viguier** [comment]
@jolelievre actually, id_tax_rules_group is always 0 in carrier table, so this method always return false.

**Comment by jf-viguier** [comment]
@jolelievre I've fixed the getIdTaxRulesGroupMostUsed sql. It's used in AdminCarrierWizardController.php.

**Comment by jf-viguier** [comment]
It's seems that I've fixed some CRLF in Carrier.php

**Review by Progi1984 — DISMISSED** [review]

**Review by matthieu-rolland — COMMENTED** [review]
seems like there are conflicts to be solved, otherwise LGTM 👍

**Comment by jf-viguier** [comment]
@matthieu-rolland conflicts resolved

**Comment by jf-viguier** [comment]
@Progi1984 @matthieu-rolland what's wrong with tests ?

**Comment by matks** [comment]
> @Progi1984 @matthieu-rolland what's wrong with tests ?

prestashop.com is down so tests are in another dimension 😅

**Comment by Julievrz** [comment]
Looks like there's no wording in this PR, I'm removing the "waiting for wording" label. Feel free to ping me if needed. 🤗

**Comment by matks** [comment]
Hi @jf-viguier sorry for coming back late. Unfortunately in September prestashop.com was down so it messed up the CI. I close and reopen the PR to reset the CI and know whether tests pass however you'll need to rebase the PR.

**Comment by jf-viguier** [comment]
@matks rebase done, have a good friday.

**Comment by jf-viguier** [comment]
@matks I've add the sql part also here https://github.com/jf-viguier/autoupgrade/pull/1

File: classes/Carrier.php [hunk 2]
```diff
@@ -599,15 +599,15 @@ public static function getIdTaxRulesGroupMostUsed()
...
+			FROM (
+				SELECT COUNT(*) n, c.id_tax_rules_group
+				FROM ' . _DB_PREFIX_ . 'carrier_tax_rules_group_shop c
+				JOIN ' . _DB_PREFIX_ . 'tax_rules_group trg ON (c.id_tax_rules_group = trg.id_tax_rules_group)
+				WHERE trg.active = 1 AND trg.deleted = 0
+				GROUP BY c.id_tax_rules_group
+				ORDER BY n DESC
+				LIMIT 1
+			) most_used'
         );
```
Comment (PierreRambaud) [inline classes/Carrier.php]: Be careful, you replace all spaces with tabs for no reason.

Comment (jf-viguier) [reply classes/Carrier.php]: @PierreRambaud it's fixed.

**Review by Progi1984 — DISMISSED** [review]
@jf-viguier Could you move the SQL query to `autoupgrade` module ?

**Comment by PierreRambaud** [comment]
Hi @jf-viguier ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Comment by jf-viguier** [comment]
@PierreRambaud what are you waiting for ? I've done the pr you asked ?
https://github.com/jf-viguier/autoupgrade/pull/1

**Comment by jf-viguier** [comment]
@PierreRambaud my bad, the pr is there https://github.com/PrestaShop/autoupgrade/pull/437 (wrong depot)

**Comment by jf-viguier** [comment]
@Progi1984 asked pr is done.

**Comment by PierreRambaud** [comment]
I reopen this PR, but next time, don't forget to answer maintainers' messages and link to the right repository :wink:

**Comment by PierreRambaud** [comment]
Anyway the PR still need to be rebased

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]
Looks OK but we need to be extra careful about validating this change. PrestaShop is not friendly with such changes; we might break something in an unknown location.

@jf-viguier can you please provide a "How to test" a little bit more complete for QA team ? it could mention places in the software where id_tax_rules_group might be used. It must be for them a path they can follow to look for issues.

**Comment by jf-viguier** [comment]
@matks this column is not used anymore since 1.7

**Comment by matks** [comment]
@jf-viguier Do you mean there is, according to our knowledge, zero line of code that do use this column in prestashop ?

**Comment by jf-viguier** [comment]
@matks yes this column is not used anymore since 1.7, The only place where it's used is in this stat method Carrier::getIdTaxRulesGroupMostUsed(). I've fixed it n the pr: this stat always return zero.
I've updated the "how to test".

**Comment by khouloudbelguith** [comment]
Hi @jf-viguier,

It is ok :heavy_check_mark: 
1. I created some carriers with different ranges
2. I created a carrier with a tax breakdown
3. I created some orders from the FO & BO 
4. I created a cart rule "free shipping" & apply it
5. I edited some orders when "Recalculate shipping costs after editing the order" is enabled & disabled

=> ok :heavy_check_mark: 
https://watch.screencastify.com/v/7T2BNIjZYutavq0jQklP

**With PR**
image

**Without PR**
image

Thanks!

**Comment by Progi1984** [comment]
Thanks @jf-viguier & @khouloudbelguith

**Comment by jf-viguier** [comment]
Thanks @Progi1984, don't forget this related pr here to delete the column https://github.com/PrestaShop/autoupgrade/pull/437

**Pull request merged**